Gir.toml

```diff
@@ -72,6 +94,14 @@ status = "generate"
     name = "run"
     ignore = true
 
+[[object]]
+name = "Gio.DataInputStream"
+status = "generate"
+    [[object.function]]
+    name = "read_line"
+    #return vec of char*
+    ignore = true
```

Comment:
That should just return a `Vec<u8>` and I thought gir handles that by now (note that the element type is set to guint8 in the gir file, it's not utf8).

Reply:
Gir generated good looking code:
```rust
    fn read_line<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P) -> Result<(Vec<u8>, usize), Error> {
        let cancellable = cancellable.into();
        let cancellable = cancellable.to_glib_none();
        unsafe {
            let mut length = mem::uninitialized();
            let mut error = ptr::null_mut();
            let ret = ffi::g_data_input_stream_read_line(self.to_glib_none().0, &mut length, cancellable.0, &mut error);
            if error.is_null() { Ok((FromGlibPtrContainer::from_glib_full(ret), length)) } else { Err(from_glib_full(error)) }
        }
    }
```
but compiling get 2 errors:
```
error[E0277]: the trait bound `u8: glib::translate::Ptr` is not satisfied
   --> src\auto\data_input_stream.rs:145:38
    |
145 |             if error.is_null() { Ok((FromGlibPtrContainer::from_glib_full(ret), length)) } else { Err(from_glib_full(error)) }
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `glib::translate::Ptr` is not implemented for `u8`
    |
    = note: required because of the requirements on the impl of `glib::translate::FromGlibPtrContainer<u8, *mut u8>` for `std::vec::Vec<u8>`
    = note: required by `glib::translate::FromGlibPtrContainer::from_glib_full`

error[E0277]: the trait bound `u8: glib::translate::FromGlibPtrArrayContainerAsVec<u8, *mut u8>` is not satisfied
   --> src\auto\data_input_stream.rs:145:38
    |
145 |             if error.is_null() { Ok((FromGlibPtrContainer::from_glib_full(ret), length)) } else { Err(from_glib_full(error)) }
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibPtrArrayContainerAsVec<u8,
 *mut u8>` is not implemented for `u8`
    |
    = note: required because of the requirements on the impl of `glib::translate::FromGlibPtrContainer<u8, *mut u8>` for `std::vec::Vec<u8>`
    = note: required by `glib::translate::FromGlibPtrContainer::from_glib_full`
```

Reply:
Hmm, possibly a bug in glib/translate.rs. Exactly that impl should be used here and should be usable, unclear why it isn't. The `glib::translate::Ptr` requirement looks dubious.

Can you create a ticket?

Reply:
Added https://github.com/gtk-rs/glib/issues/275